app/models/provider.rb

```diff
@@ -13,5 +13,9 @@
 #
 class Provider < ApplicationRecord
   validates :provider_code, presence: true
-  validates :provider_code, uniqueness: { case_sensitive: false }
+  validates :provider_code,
+            uniqueness: {
+              case_sensitive: false,
+              message: "Provider already exists!"
```

Comment:
I'm sure our (soon to join) content designer will want to configure custom/meaningful error messages.

However, as a general rule of thumb:

1. I don't think we should use exclamation marks in error messages. It feels a bit shouty!!! 🙃
2. I think we should aim to [use locale files to set custom validation messages](https://guides.rubyonrails.org/i18n.html#error-message-scopes) rather than hardcoding them in the models. I believe this same message could be set using the locale key `en.activerecord.errors.models.provider.attributes.provider_code.taken`.

Localised error messages give us more flexibility if we want to [customise the full message](https://edgeguides.rubyonrails.org/configuring.html#configuring-active-model) further down the line. For example, as was needed in [a previous Rails project I worked on](https://github.com/alphagov/whitehall/pull/8361).

Reply:
resolved in [3b78a77](https://github.com/DFE-Digital/itt-mentor-services/pull/48/commits/3b78a77c049f64ecf0a81a96fc412ccf8260cce1)